RecoMTD/DetLayers/test/MTDRecoGeometryAnalyzer.cc

```diff
@@ -31,27 +32,28 @@
 #include <DataFormats/ForwardDetId/interface/BTLDetId.h>
 #include <DataFormats/ForwardDetId/interface/ETLDetId.h>
 
+#include <memory>
 #include <sstream>
 
 #include "CLHEP/Random/RandFlat.h"
 
 using namespace std;
 using namespace edm;
 
-class MTDRecoGeometryAnalyzer : public EDAnalyzer {
+class MTDRecoGeometryAnalyzer : public global::EDAnalyzer<> {
```

Comment:
this module is using `CLHEP::RandFlat::shoot` and I do not see `HepRandomEngine`
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMRandomNumberGeneratorService
I'm not sure it's a good idea to have a `global` module in this case.

Reply:
I thought about this, but if you look here:
https://gitlab.cern.ch/CLHEP/CLHEP/-/blob/develop/Random/src/Random.cc

You see the following line of code, which makes the engines thread local
(unless CLHEP_THREAD_LOCAL is somewhere  defined as not thread local
and I think other things would be broken badly if that were the case).

```
static CLHEP_THREAD_LOCAL defaults* theDefaults = defaultsForAllThreads.createNewDefaults();
```

If the engines are thread locals. there is no data race issue. It's OK
for them to be used by global modules.

This test module is not using the random number service. That would
break replay and it means it's not getting seeds from the service, but that
is all independent of whether or not this should be global or one.
It could have issues with the seeds not generating independent
sequences...

That said, I could change it to "one" just to avoid thinking about this.
The test modules are used in only one single threaded test. At some
level, it doesn't matter which way we go.

Reply:
To be clear, this is not supposed to enter any production workflow, it is just a standalone test. The random numbers are not supposed to compete with any other random number source as far as I can see

Reply:
Yes. Given that the module is only used in a standalone unit test, it isn't worth the effort to fix it to use the random service...

If someone still prefers this to be a "one", let me know and I will change it (it wouldn't take more than a few minutes). I'm most interested in getting the PR merged quickly and don't really have a preference whether it is "one" or "global". My main concern is getting the unit test to pass.